Objective-C/CBLReplicatorConfiguration.m

```diff
@@ -215,6 +247,12 @@ - (NSDictionary*) effectiveOptions {
     if (_heartbeat > 0)
         options[@kC4ReplicatorHeartbeatInterval] = @(_heartbeat);
     
+    if (_maxRetryWaitTime > 0)
+        options[@kC4ReplicatorOptionMaxRetryInterval] = @(_maxRetryWaitTime);
+    
+    if (_maxRetries >= 0)
+        options[@kC4ReplicatorOptionMaxRetries] = @(_maxRetries);
```

Comment:
_maxRetries is -1 by default so it needs to set to an appropriate value for continuous or single shot here.